### PR TITLE
Improve ci-db-backcompat.sh for use with souregraph.com

### PR DESCRIPTION
**Pull request by efritz**
Closes https://github.com/sourcegraph/sourcegraph/issues/11797.

### CODE REVIEW

**Comment by efritz** [comment]
@sourcegraph/distribution Wondering if it is possible to give the current buidkite user access to list pods on sourcegraph.com: 

https://buildkite.com/sourcegraph/sourcegraph/builds/67133#b46e2ce2-382f-4295-8d6d-4a1f415ea679/75-78

Or if there is an alternate way to get the current deployed commit version you'd suggest using instead.

File: dev/deployed-commit.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# This script determines the commit of sourcegraph/sourcegraph that is currently
+# running on sourcegraph.com.
+
+set -e
+
+# Determine the image that is currently running in the k8s prod context. This will
+# return the last sourcegraph/frontend image sorted by age (i.e., the newest pod).
+IMAGE=$(
+  kubectl -n prod get pods -l app=sourcegraph-frontend --sort-by=.metadata.creationTimestamp -o jsonpath='{..image}' |
```
Comment (pecigonzalo) [inline dev/deployed-commit.sh:11]: Is there a way for us to get this information without access to the prod cluster? eg: using the docker tag, or some other metedata.

Comment (efritz) [reply dev/deployed-commit.sh:11]: This gets the docker tag. We could alternatively check what's currently configured in deploy-sourcegraph-dot-com.

Comment (pecigonzalo) [reply dev/deployed-commit.sh:11]: Yes, but I believe getting it from the prod cluster is not a good idea, as it requires CI (a dangerous thing IMO)  to have access to prod for a simple CI check.
Maybe someone in @sourcegraph/distribution knows if we push a special tag to image on deploy to sourcegraph.com or similar, maybe we should add that metadata somewhere.

Out of scope, but in many cases its a good idea to push extra tags to images that are used as metadata, for example pushing a `com` tag to the current image that we use for sourcegraph.com. Or a `v1` to the latest `v1.x` image.
While we should not deploy from those tags directly, we can use it to discover the sha or leverage them for CI.

Alternatively, we could push that information to any other location that CI can access that is not as sensitive, like GCP project-wide metadata.

Comment (tsenart) [reply dev/deployed-commit.sh:11]: `curl https://sourcegraph.com/__version`

File: dev/deployed-commit.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# This script determines the commit of sourcegraph/sourcegraph that is currently
+# running on sourcegraph.com. The returned version has one of the following
+# formats:
+#
+#   - v{semver}
+#   - {buildkite build num}_{date}_{abbreviated commit}
+#
+# In either case, the following cut will give us something parseable by rev-parse.
+
+git rev-parse "$(curl https://sourcegraph.com/__version | cut -d'_' -f3)"
```
Comment (pecigonzalo) [inline dev/deployed-commit.sh:12]:
```suggestion
set -o pipefail
git rev-parse "$(curl -sf https://sourcegraph.com/__version | cut -d'_' -f3)"
```

I would add add `pipefail` and [-sf](https://explainshell.com/explain?cmd=curl+-sf)

**Pull request merged**